app/javascript/components/Comment.js

```diff
@@ -0,0 +1,140 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class Comment extends React.Component {
+  render() {
+    const {  
+      authorPicUrl,
+      authorPicFilename,
+      authorUsername,
+      cid,
+      commentName,
+      htmlCommentText,
+      rawCommentText,
+      replies,
+      replyTo,
+      timeCreatedString
+    } = this.props.comment;
+
+    let authorSection = [];
```

Comment:
This is really nice. We might also store some of these subsections (strings) in a templates subfolder. I'm not familiar with React conventions though, is that often done to make things more concise and readable?

Reply:
Yes, the main thing this is doing is for readability. There's a brief-ish writeup [here](https://www.freecodecamp.org/news/the-basics-of-destructuring-props-in-react-a196696f5477/).

Basically I 'destructure' the props here so that I can just refer to `replies` instead of `this.props.comment.replies`, as well as all the other props. Just saves effort typing, makes things more readable, and forces me to be clear about what props belong to, and need to be present in, whichever component.